controller/scheduler/scheduler.go

```diff
@@ -272,20 +273,20 @@ func (s *Scheduler) streamHostEvents() error {
 				}
 			}
 			log.Warn("host event stream disconnected", "err", stream.Err())
-			for {
-				if err := connect(); err == nil {
-					continue outer
-				}
-				time.Sleep(100 * time.Millisecond)
-			}
 		}
 	}()
 
-	select {
-	case <-current:
-		return nil
-	case <-time.After(30 * time.Second):
-		return errors.New("timed out waiting for current host list")
+	// wait until we have the current list of hosts and their
+	// jobs before starting the main scheduler loop
+	start := time.Now()
+	tick := time.Tick(30 * time.Second)
```

Comment:
This is not actually correct, since the ticker is on a timer heap internally, you need to call `Stop` to remove it from the heap so that it can be garbage collected.